core/backend.ml

```diff
@@ -42,21 +52,41 @@ let perform_for_side_effects side_effecting_transformer tyenv p =
 module Pipelines =
 struct
 
-    let optimisation_pipeline = [
+    let optimisation_pipeline () = [
+        debug_tell "optimising IR";
         IrTraversals.ElimDeadDefs.program;
         IrTraversals.Inline.program;
+        debug_tell "optimised IR"
       ]
 
-    let typechecking_pipeline = [
+    let simplify_type_structure_program () = [
+        debug_tell "simplifying types";
         (*IrTraversals.NormaliseTypes.program;
         IrTraversals.ElimRecursiveTypeCycles.program;*)
         IrTraversals.ElimTypeAliases.program;
         IrTraversals.ElimBodiesFromMetaTypeVars.program;
         (only_if_set Basicsettings.Ir.show_compiled_ir_after_backend_transformations print_program);
+        debug_tell "simplified types"
```

Comment:
I would move this debug output before `(only_if_set ...` line. Right now the IR is displayed between `"simplifying types"` and `"simplified types"` messages so it's not clear whether this is IR before or after transformations.

Reply:
I disagree, this follows the scheme for `show_compiled_ir`.